openstack/tools/func_test_tools/common.py

```diff
@@ -56,13 +56,11 @@ def get_branch_jobs(self, branch, project_templates):
 class OSCIConfig():
     """ Extract information from osci.yaml """
     def __init__(self):
-        if not os.path.exists('osci.yaml'):
-            self._osci_config = {}
-        else:
-            with open('osci.yaml', encoding='utf-8') as fd:
-                self._osci_config = yaml.safe_load(fd)
+        path = os.path.join(os.environ.get('CHARM_ROOT_PATH', ''), 'osci.yaml')
+        with open(path, encoding='utf-8') as fd:
```

Comment:
When opening a file we should have in an exception catcher so that we can handle it properly

Something like:
```
try:
    with open(path, encoding='utf-8') as fd:
        self._osci_config = yaml.safe_load(fd)
except Exception as e:
    print(f"Error opening {file}: {e}")

Reply:
no i actually want it to fail hard since if it cant find this information it cant do its job properly (which is what happened before)